lib/components/form/call-taker/date-time-picker.tsx

```diff
@@ -121,69 +151,32 @@ const DateTimeOptions = ({
   )
   const [date, setDate] = useState<string | undefined>(initialDate)
   const [time, setTime] = useState<string | undefined>(initialTime)
-  const [typedTime, setTypedTime] = useState<string | undefined>(initialTime)
+  const [typedTime, setTypedTime] = useState<string | undefined>(
+    safeFormat(parseInputAsTime(homeTimezone, time, date), timeFormat, {
+      timeZone: homeTimezone
+    })
+  )
 
   const timeRef = useRef(null)
 
   const intl = useIntl()
 
-  /**
-   * Parse a time input expressed in the agency time zone.
-   * @returns A date if the parsing succeeded, or null.
-   */
-  const parseInputAsTime = (
-    timeInput: string = getCurrentTime(homeTimezone),
-    date: string = getCurrentDate(homeTimezone)
-  ) => {
-    if (!timeInput) timeInput = getCurrentTime(homeTimezone)
-
-    // Match one of the supported time formats
-    const matchedTimeFormat = SUPPORTED_TIME_FORMATS.find((timeFormat) =>
-      isMatch(timeInput, timeFormat)
-    )
-    if (matchedTimeFormat) {
-      const resolvedDateTime = format(
-        parse(timeInput, matchedTimeFormat, new Date()),
-        'HH:mm:ss'
-      )
-      return toDate(`${date}T${resolvedDateTime}`)
-    }
-    return ''
-  }
-
-  const dateTime = parseInputAsTime(time, date)
+  const dateTime = parseInputAsTime(homeTimezone, time, date)
 
   // Update state when external state is updated
   useEffect(() => {
     if (initialDate !== date) setDate(initialDate)
     if (initialTime !== time) {
-      setTime(initialTime)
+      handleTimeChange(initialTime || '')
     }
   }, [initialTime, initialDate])
 
-  useEffect(() => {
-    // Don't update if still typing
-    if (timeRef.current !== document.activeElement) {
-      setTypedTime(
-        safeFormat(dateTime, timeFormat, {
-          timeZone: homeTimezone
-        }) ||
-          // TODO: there doesn't seem to be an intl object present?
-          'Invalid Time'
-      )
-    }
-  }, [time])
-
   useEffect(() => {
     if (initialDepartArrive && departArrive !== initialDepartArrive) {
       setDepartArrive(initialDepartArrive)
     }
   }, [initialDepartArrive])
```

Comment:
While we are in fixing dependency arrays, could you adding `departArrive` in here too per eslint.

Reply:
same as above, causes a loop